hips/hip-9999.md

```diff
@@ -0,0 +1,92 @@
+
+With the `import-values` directive, there is a mechanism in Helm to use the default values of a subchart as a value in the parent chart and define the name under which the subchart value is available in the parent chart.
+This HIP describes the `export-values` directive which is similar to the [`import-values`](https://helm.sh/docs/topics/charts/#importing-child-values-via-dependencies) directive, and allows exporting values from the parent chart to the subchart and defining the name under which it should be available.
+With that the values of the parent chart and the values of the subcharts can be defined independently. Subcharts can use their local values exclusively and don't have to be aware of the usage as a subchart. Parent charts on the other hand, can provide a consistent configuration api to their users and orchestrate the exchange of values between the subcharts using `export-values`, `import-values` or a combination thereof.
+
+This change was discussed in [helm/#3242]. There were several attempts to implement this: [helm/#7477], [helm/#10059].
+
+## Specification
+
+With the proposed `export-values` directive, useres are able to specify in the **parent** chart (snippet below) which and where values should be exported to the **subchart**.
+
+```yaml
+# Chart.yaml:
+dependencies:
+  - name: client
+    version: 1.0.0
+    export-values:
+    - parent: "port"
+      child: "port"
+  - name: server
+    version: 1.0.0
+    export-values:
+    - parent: "port"
+      child: "port"
+```
+
+```yaml
+# values.yaml
+port: 8080
+```
+
+### Desired Evaluation Order
+
+1. Composite chart receives values
+   1. Explicitly set via `--set KEY=VALUE` flag(s)
+   1. Explicitly set via `--values` flag(s)
+   1. Chart defaults (`values.yaml`)
+1. Subchart receives values
+
+### Value Precedence
+
+1. Explicitly set for composit value (`port`)
+1. Explicitly set for subchart value (`client.port`)
+1. default for composit value (`values.yaml` in composite chart)
+1. default for subchart value (`values.yaml` in subchart)
+
+### Combination with `import-values`
+
+It should be able to use both, `export-values` and `import-values`, indepentently and in combination. It should also be possible to export a value which was imported from third chart.
+
+## Motivation
+
+Currently, it is not possible to create a composite chart without leaking the structure of the values of the subcharts used. This would allow to freely structure the values the user interacts with.
```

Comment:
```suggestion
Currently, it is not possible to compose charts using dependencies without leaking the structure of the values of the subcharts used. This would allow to freely structure the values the user interacts with.
```

Also, we are maybe a bit this on the motivational aspects. Maybe some part I proposed in the actract should be reiterated here? IDK